File: libs/power-log-parser/src/lib/state/player-report.ts
import { GameTag } from '@firestone-hs/reference-data';
import type { GameState } from './game-state';

export class PlayerReport {
	TotalHealth: number = 0;
	DamageTaken: number = 0;
	ArmorLeft: number = 0;

	static buildPlayerReport(state: GameState, id: number): PlayerReport {
		const playerState = state.GetPlayerHeroEntity(id)!;
		const report = new PlayerReport();
		report.TotalHealth = playerState.GetTag(GameTag.HEALTH);
		report.DamageTaken = playerState.GetTag(GameTag.DAMAGE, 0);
		report.ArmorLeft = playerState.GetTag(GameTag.ARMOR, 0);
		return report;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Events/PlayerReport.cs
#region
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using HearthstoneReplays.Parser.ReplayData.GameActions;
#endregion

namespace HearthstoneReplays.Parser.ReplayData.Entities
{
	public class PlayerReport
    {
        public int TotalHealth { get; set; }
        public int DamageTaken { get; set; }
        public int ArmorLeft { get; set; }

        internal static PlayerReport BuildPlayerReport(GameState state, int id)
        {
            var playerState = state.GetPlayerHeroEntity(id);
            return new PlayerReport
            {
                TotalHealth = playerState.GetTag(GameTag.HEALTH),
                DamageTaken = playerState.GetTag(GameTag.DAMAGE, 0),
                ArmorLeft = playerState.GetTag(GameTag.ARMOR, 0)
            };
        }
    }
}